code/modules/ruins/objects_and_mobs/ash_walker_den.dm

```diff
@@ -38,15 +38,33 @@
 /obj/structure/lavaland/ash_walker/proc/consume()
 	for(var/mob/living/H in view(src, 1)) //Only for corpse right next to/on same tile
 		if(H.stat)
-			visible_message("<span class='warning'>Serrated tendrils eagerly pull [H] to [src], tearing the body apart as its blood seeps over the eggs.</span>")
-			playsound(get_turf(src),'sound/magic/demon_consume.ogg', 100, TRUE)
 			for(var/obj/item/W in H)
 				if(!H.dropItemToGround(W))
 					qdel(W)
+			if(issilicon(H)) //no advantage to sacrificing borgs...
+				H.gib()
+				visible_message("<span class='notice'>Serrated tendrils eagerly pull [H] apart, but find nothing of interest.</span>")
+				return
+			if(H.mind?.has_antag_datum(/datum/antagonist/ashwalker) && H.get_ghost(FALSE, TRUE)) //special interactions for dead lava lizards with ghosts attached
+				visible_message("<span class='warning'>Serrated tendrils carefully pull [H] to [src], absorbing the body and creating it anew.</span>")
+				H.notify_ghost_cloning(message = "Your body has been returned to the nest. You are being remade anew!", sound = 'sound/magic/enter_blood.ogg')
+				addtimer(CALLBACK(src, .proc/remake_walker, H.mind, H.real_name), 5 SECONDS)
+				new /obj/effect/gibspawner/generic(get_turf(H))
+				qdel(H)
```

Comment:
this won't delete any gear that was on the ashwalker's old body, right?

Reply:
No, one of the first things the tendril does is strip bodies of all items.